src/main/java/org/java_websocket/client/WebSocketClient.java

```diff
@@ -452,6 +453,15 @@ public void run() {
 			onWebsocketError( engine, e );
 			engine.closeConnection( CloseFrame.NEVER_CONNECTED, e.getMessage() );
 			return;
+		} catch (InternalError e) {
+			// https://bugs.openjdk.java.net/browse/JDK-8173620
+			if (e.getCause() instanceof InvocationTargetException && e.getCause().getCause() instanceof IOException) {
+				IOException cause = (IOException) e.getCause().getCause();
+				onWebsocketError(engine, cause);
+				engine.closeConnection(CloseFrame.NEVER_CONNECTED, cause.getMessage());
+				return;
+			}
+			throw e;
```

Comment:
Why re-throw InternalError instead of returning normally?

Reply:
The idea was to handle only encapsulated `IOException` by calling the `onError` of the `WebSocketClient` and closing the connection as if we received the `IOException` directly.
This way the implementation of the `WebSocketClient` can handle the error.

If it's any other kind of InternalError we should not handle it here, the same way we don't usually include catching `Error` in all `try{...} catch{...}`

What do you think?

Reply:
You're completely right, somehow didn't notice the 'return' statement within 'if'